### PR TITLE
Changed type for some properties to Color from Property

### LINKED ISSUE #9667: Typescript errors when setting color properties of various Cesium objects
When setting a color-property (color, fillColor, outlineColor, etc) of an instance of a Cesium object, the typescript definitions don't seem to match the actual implementation in CesiumJS.
For instance, if I try to set the fillColor of a LabelGraphics object:

`labelGraphics.fillColor = Cesium.Color.PURPLE`

typescript errors occur when setting the property to a Cesium.Color directly. Wrapping the Color in a ColorMaterialProperty:

`labelGraphics.fillColor = new ColorMaterialProperty(Cesium.Color.PURPLE)`

this silences the typescript error, but the color isn't applied during runtime.
If I ignore the typescript error and set the color directly it does work, but then the typescript errors interfere with my normal development process.

Sandcastle example:
AFAIK there is no option to use typescript in sandcastle, so it's not possible to repro this issue there.
Browser: Any

Operating System: MacOS (not likely to have a bearing on this)

Looking at the typescript definitions in node_modules/cesium/Source/Cesium.d.ts, I see at line 21534:

`  fillColor: Property | undefined;`

If I change this to:

`  fillColor: Color | Property | undefined;`

this solves the problem, but it's debatable whether this is the proper solution or rather that the implementation in CesiumJS should be fixed to match the typescript definitions.

### PR DESCRIPTION
**Pull request by markchagers**
Fixes #9667. Some color properties of several Graphics classes (i.e. LabelGraphics.outlineColor) accept only a Cesium.Color, trying to set a ColorMaterialProperty has no effect at runtime. However the typescript definitions specify using a Property. This pull request attempts to address this by changing the type annotations for these properties. Going the other route: changing the implementation to work properly with a ColorMaterialProperty may be a better solution eventually, but is currently beyond my expertise.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thank you so much for the pull request @markchagers! I noticed this is your first pull request and I wanted to say welcome to the Cesium community!

The [Pull Request Guidelines](https://github.com/CesiumGS/cesium/blob/master/CONTRIBUTING.md#pull-request-guidelines) is a handy reference for making sure your PR gets accepted quickly, so make sure to skim that.

* :x: Missing `CONTRIBUTORS.md` entry.
    * Please add yourself to the [contributors](https://github.com/markchagers/cesium/blob/fix/typescript-definitions-color-members/CONTRIBUTORS.md) file!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/CesiumGS/cesium/blob/fix/typescript-definitions-color-members/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.
* :grey_question: Unit tests were not updated.
    * Make sure you've [updated tests](https://github.com/CesiumGS/cesium/tree/master/Documentation/Contributors/TestingGuide) to reflect your changes, added tests for any new code, and ran the code coverage tool.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

**Comment by markchagers** [comment]
Please let me know if there is anything else that needs to be done before this PR can be evaluated. I have not changed or extended any tests because I'm not aware that any of these changes require it.

**Comment by srothst1** [comment]
@markchagers thank you for the pull request! We will look over this ASAP - I tagged a few other Cesium developers for a review.

Fixes #9667. This pull request stems from the following [community forum thread](https://community.cesium.com/t/typescript-errors-when-setting-color-of-pointgraphics/14332).

@lilleyse 
@ebogo1

File: CHANGES.md [hunk 1]
```diff
@@ -10,6 +14,8 @@
 - Fixed the ability to set a material's image to `undefined` and `Material.DefaultImageId`. [#9644](https://github.com/CesiumGS/cesium/pull/9644)
 - Fixed render crash when creating a `polylineVolume` with very close points. [#9669](https://github.com/CesiumGS/cesium/pull/9669)
 
+=======
```
Comment (ebogo1) [inline CHANGES.md:17]: This line isn't needed
```suggestion
```

File: CHANGES.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
 # Change Log
 
+##### Changes PR https://github.com/CesiumGS/cesium/pull/9673
+
+- Changes type annotations for several color properties from Property|undifined to Color|undefined
```
Comment (ebogo1) [inline CHANGES.md:5]:
```suggestion
- Changes type annotations for several color properties from `Property|undefined` to `Color|undefined`
```

**Review by ebogo1 — COMMENTED** [review]
Thanks @markchagers! This looks fine to me, just a couple nits in CHANGES.md. 

@mramato Any objections to this change?

**Comment by mramato** [comment]
> (i.e. LabelGraphics.outlineColor) accept only a Cesium.Color,

This is incorrect. These are `Property` instances, just not a `MaterialProperty`, i.e. 

```js
graphics.outlineColor = new ConstantProperty(Color.BLUE);
graphics.outlineColor = new SampledProperty(Color);
```

`graphics.outlineColor = Color.BLUE;` works but will result in `graphics.outlineColor` being a `ConstantProperty` (using implicit conversion borrowed from C#). Because of limitations in TypeScript, we can't tag properties as being multiple types. See this comment for some additional details: https://github.com/CesiumGS/cesium/issues/8898#issuecomment-637579223

If TypeScript has improved and we can better express implicity conversion, that would be awesome.

**Comment by markchagers** [comment]
@mramato your statement:
> Because of limitations in TypeScript, we can't tag properties as being multiple types. 
I don't get this, after all, we can do this:
`someProp: Property | Color | undifined`
Or is that not what you meant?

**Comment by markchagers** [comment]
@mramato I'm updating the Cesium lib on a project I inherited. I the course of this updat I run into this issue:
`this.cesiumEntity.label.fillColor = Cesium.Color.BLUE;`
cesiumEntity is a Cesium entity with a LabelGraphics member. In the above line of code the color is updated to a different color. This used to work in Cesium 1.55 (yeah, I know).
This line is flagged by typescript (in VSCode) as incorrect: "Type 'Color' is missing the following properties from type 'Property': ...".
But when I changed the line to:
`this.cesiumEntity.label.fillColor = new Cesium.ColorMaterialProperty(Cesium.Color.BLUE);`
It didn't work (ColorMaterialProperty was one of the types suggested by VSCode intellisense and judging by the name it seemed a good fit). However, I can confirm that using a ConstantProperty does work, so my pull request is unnecessary.

**Pull request closed without merging**